.github/workflows/main.yml

```diff
@@ -11,13 +11,35 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-      - name: Cache cargo
+
+      - name: Detect the installed Rust version
+        id: rustc
+        run: echo "::set-output name=version::$(rustc -V)"
+
+      - name: Cache Cargo's registry cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/cache
+          key: cargo-registry-cache-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: |
+            cargo-registry-cache-
+
+      - name: Cache Cargo's registry index
+        uses: actions/cache@v2
+        with:
+          path: ~/.cargo/registry/index
+          key: cargo-registry-index-${{ hashFiles('**/Cargo.lock') }}
```

Comment:
Why key this on Cargo.lock? Even if the index is partially out of date, it will still be useful, right? Or is the idea to cache the latest version of the index possible?

Reply:
Yes, the goal is to cache the most recent clone of the index, to avoid having to clone it at all during a build. If the `Cargo.lock` is updated, the `restore-keys` will download the most recent cache available.